src/components/ActionButton/ActionButton.tsx

```diff
@@ -95,23 +94,29 @@ const ActionButton = ({
 
     if (!loading && showLoader) {
       const now = new Date();
-      // calculate elasped time of loading from user and do subtraction to LOADER_MIN_DURATION,
-      // also add an edge case when time diff is less than 0 to be 0.
+      // Keep track of time when loading is occured.
       const loadingMilliseconds: number =
         now.getTime() - (startLoadTime.current ?? now).getTime();
-      loadingTimeTracker.current = Math.max(
+
+      // also add an edge case when time diff is less than 0 to be 0.
+      const loadingTimeout = Math.max(
```

Comment:
On a second thought, since we have the timeout itself called `loaderTimeout`, I think calling the duration of the timeout `loadingTimeout` is a bit confusing. Could you please rename this from `loadingTimeout` to `timeoutDuration`?